examples/DotNetCore/WebDemoWithEventHub/WebDemoWithEventHub/EventHubService.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs;
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Processor;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration;
+using Microsoft.Extensions.Logging;
+
+namespace WebDemoWithEventHub
+{
+    public class EventHubService
+    {
+        private IConfigurationRefresher _refresher;
+
+        private ILogger<EventHubService> _logger;
+
+        public EventHubService(IConfiguration config, IConfigurationRefresherProvider providers, ILogger<EventHubService> logger)
+        {
+            _logger = logger;
+
+            EventHubConnection eventHubConnection = config.GetSection("WebDemo:EventHubConnection").Get<EventHubConnection>();
+            InitEventHubProcessor(eventHubConnection);
+
+            _refresher = providers.Refreshers.FirstOrDefault(r => r is IConfigurationRefresher);
```

Comment:
Why `r => r is IConfigurationRefresher`?

Reply:
refresherProvider.Refreshers is an IEnumerable. I'm trying to assign the first IConfigurationRefresher instance to _configuartionRerfresher.

---
In reply to: [464787077](https://github.com/Azure/AppConfiguration/pull/329#discussion_r464787077) [](ancestors = 464787077)

Reply:
Understood. My question is why the lambda is needed.

---
In reply to: [466011776](https://github.com/Azure/AppConfiguration/pull/329#discussion_r466011776) [](ancestors = 466011776,464787077)

Reply:
I thought it would be simpler. Would a forloop be clearer to understand?

---
In reply to: [466102871](https://github.com/Azure/AppConfiguration/pull/329#discussion_r466102871) [](ancestors = 466102871,466011776,464787077)

Reply:
Won't just `FirstOrDefault()` be sufficient?

---
In reply to: [466813876](https://github.com/Azure/AppConfiguration/pull/329#discussion_r466813876) [](ancestors = 466813876,466102871,466011776,464787077)

Reply:
Changed to FirstOrDefault()

---
In reply to: [467235445](https://github.com/Azure/AppConfiguration/pull/329#discussion_r467235445) [](ancestors = 467235445,466813876,466102871,466011776,464787077)